### PR TITLE
Some quick and minor fixes in code quality

### PR DESCRIPTION
**Pull request by shivamvats**
@certik

### CODE REVIEW

File: sympy/polys/ring_series.py [hunk 1]
```diff
@@ -1638,7 +1642,7 @@ def rs_tanh(p, x, prec):
                 const = R(tanh(c))
             except ValueError:
                 raise DomainError("The given series can't be expanded in this "
-                "domain")
+                    "domain")
```
Comment (leosartaj) [inline sympy/polys/ring_series.py]: It is still under-indented.

File: sympy/polys/ring_series.py [hunk 2]
```diff
@@ -1509,7 +1513,7 @@ def rs_atanh(p, x, prec):
                 const = R(atanh(c))
             except ValueError:
                 raise DomainError("The given series can't be expanded in this "
-                "domain")
+                    "domain")
```
Comment (leosartaj) [inline sympy/polys/ring_series.py]: It is still under-indented.

**Comment by leosartaj** [comment]
I ran a quick check with flake8. There are a lot of errors. You should fix those as the PR is aimed at code quality.

File: sympy/polys/ring_series.py (hunk 1 above)
Comment (certik) [inline sympy/polys/ring_series.py]: @leosartaj can you be more specific? It seems to be using 4 spaces now.

**Comment by certik** [comment]
Looks good to me. @leosartaj had a comment, it's fine with me either way.

File: sympy/polys/ring_series.py (hunk 1 above)
Comment (leosartaj) [inline sympy/polys/ring_series.py]: Output by flake8.

> sympy/polys/ring_series.py:1645:21: E128 continuation line under-indented for visual indent

By aligning ", this can be fixed.

Comment (certik) [inline sympy/polys/ring_series.py]: What should you align it to?

**Comment by shivamvats** [comment]
@leosartaj I did not do a thorough code quality test. These were some of the obvious improvements I could see. As for the errors, pylint seems to have issues with single characters as variables, I don't know about flake8. It would be great if you could point out the errors that you see.

@certik Cool!

File: sympy/polys/ring_series.py (hunk 1 above)
Comment (leosartaj) [inline sympy/polys/ring_series.py]: The " on both 1644 and 1645.

**Comment by leosartaj** [comment]
Output of ``flake8 ring_series.py``

sympy/polys/ring_series.py:1:1: F401 'ExpressionDomain' imported but unused
sympy/polys/ring_series.py:5:5: E128 continuation line under-indented for visual indent
sympy/polys/ring_series.py:8:1: F401 'Pow' imported but unused
sympy/polys/ring_series.py:10:1: F401 'S' imported but unused
sympy/polys/ring_series.py:10:1: F401 'evaluate' imported but unused
sympy/polys/ring_series.py:12:1: F401 'ceiling' imported but unused
sympy/polys/ring_series.py:15:1: F401 'wraps' imported but unused
sympy/polys/ring_series.py:51:1: E302 expected 2 blank lines, found 1
sympy/polys/ring_series.py:61:1: E302 expected 2 blank lines, found 1
sympy/polys/ring_series.py:88:1: E302 expected 2 blank lines, found 1
sympy/polys/ring_series.py:105:5: F841 local variable 'n' is assigned to but never used
sympy/polys/ring_series.py:113:1: E302 expected 2 blank lines, found 1
sympy/polys/ring_series.py:115:81: E501 line too long (97 > 80 characters)
sympy/polys/ring_series.py:137:13: F841 local variable 'num' is assigned to but never used
sympy/polys/ring_series.py:151:1: E302 expected 2 blank lines, found 1
sympy/polys/ring_series.py:153:81: E501 line too long (100 > 80 characters)
sympy/polys/ring_series.py:164:13: F841 local variable 'num' is assigned to but never used
sympy/polys/ring_series.py:227:1: E302 expected 2 blank lines, found 1
sympy/polys/ring_series.py:267:1: E302 expected 2 blank lines, found 1
sympy/polys/ring_series.py:305:13: E227 missing whitespace around bitwise or shift operator
sympy/polys/ring_series.py:314:1: E302 expected 2 blank lines, found 1
sympy/polys/ring_series.py:371:74: E502 the backslash is redundant between brackets
sympy/polys/ring_series.py:372:25: E128 continuation line under-indented for visual indent
sympy/polys/ring_series.py:379:1: E302 expected 2 blank lines, found 1
sympy/polys/ring_series.py:405:1: E302 expected 2 blank lines, found 1
sympy/polys/ring_series.py:409:15: F821 undefined name 'PoleError'
sympy/polys/ring_series.py:409:81: E501 line too long (89 > 80 characters)
sympy/polys/ring_series.py:412:1: E302 expected 2 blank lines, found 1
sympy/polys/ring_series.py:445:13: E128 continuation line under-indented for visual indent
sympy/polys/ring_series.py:489:81: E501 line too long (99 > 80 characters)
sympy/polys/ring_series.py:511:1: E302 expected 2 blank lines, found 1
sympy/polys/ring_series.py:571:9: F841 local variable 'sb' is assigned to but never used
sympy/polys/ring_series.py:577:1: E302 expected 2 blank lines, found 1
sympy/polys/ring_series.py:662:1: E302 expected 2 blank lines, found 1
sympy/polys/ring_series.py:691:1: E302 expected 2 blank lines, found 1
sympy/polys/ring_series.py:718:1: E302 expected 2 blank lines, found 1
sympy/polys/ring_series.py:732:81: E501 line too long (81 > 80 characters)
sympy/polys/ring_series.py:771:1: E302 expected 2 blank lines, found 1
sympy/polys/ring_series.py:785:1: E302 expected 2 blank lines, found 1
sympy/polys/ring_series.py:804:1: E302 expected 2 blank lines, found 1
sympy/polys/ring_series.py:846:1: E302 expected 2 blank lines, found 1
sympy/polys/ring_series.py:901:21: E128 continuation line under-indented for visual indent
sympy/polys/ring_series.py:904:45: E261 at least two spaces before inline comment
sympy/polys/ring_series.py:907:21: E128 continuation line under-indented for visual indent
sympy/polys/ring_series.py:918:1: E302 expected 2 blank lines, found 1
sympy/polys/ring_series.py:958:81: E501 line too long (83 > 80 characters)
sympy/polys/ring_series.py:959:25: E128 continuation line under-indented for visual indent
sympy/polys/ring_series.py:964:81: E501 line too long (83 > 80 characters)
sympy/polys/ring_series.py:965:25: E128 continuation line under-indented for visual indent
sympy/polys/ring_series.py:973:1: E302 expected 2 blank lines, found 1
sympy/polys/ring_series.py:1012:1: E302 expected 2 blank lines, found 1
sympy/polys/ring_series.py:1024:1: E302 expected 2 blank lines, found 1
sympy/polys/ring_series.py:1041:5: F841 local variable 'index' is assigned to but never used
sympy/polys/ring_series.py:1054:21: E128 continuation line under-indented for visual indent
sympy/polys/ring_series.py:1060:20: E128 continuation line under-indented for visual indent
sympy/polys/ring_series.py:1081:1: E302 expected 2 blank lines, found 1
sympy/polys/ring_series.py:1097:1: E302 expected 2 blank lines, found 1
sympy/polys/ring_series.py:1134:21: E128 continuation line under-indented for visual indent
sympy/polys/ring_series.py:1140:21: E128 continuation line under-indented for visual indent
sympy/polys/ring_series.py:1151:1: E302 expected 2 blank lines, found 1
sympy/polys/ring_series.py:1196:1: E302 expected 2 blank lines, found 1
sympy/polys/ring_series.py:1217:1: E302 expected 2 blank lines, found 1
sympy/polys/ring_series.py:1255:21: E128 continuation line under-indented for visual indent
sympy/polys/ring_series.py:1261:21: E128 continuation line under-indented for visual indent
sympy/polys/ring_series.py:1275:1: E302 expected 2 blank lines, found 1
sympy/polys/ring_series.py:1311:1: E302 expected 2 blank lines, found 1
sympy/polys/ring_series.py:1351:21: E128 continuation line under-indented for visual indent
sympy/polys/ring_series.py:1357:21: E128 continuation line under-indented for visual indent
sympy/polys/ring_series.py:1379:1: E302 expected 2 blank lines, found 1
sympy/polys/ring_series.py:1417:21: E128 continuation line under-indented for visual indent
sympy/polys/ring_series.py:1423:21: E128 continuation line under-indented for visual indent
sympy/polys/ring_series.py:1435:25: E203 whitespace before ','
sympy/polys/ring_series.py:1435:26: E231 missing whitespace after ','
sympy/polys/ring_series.py:1445:1: E302 expected 2 blank lines, found 1
sympy/polys/ring_series.py:1457:1: E302 expected 2 blank lines, found 1
sympy/polys/ring_series.py:1473:1: E302 expected 2 blank lines, found 1
sympy/polys/ring_series.py:1510:21: E128 continuation line under-indented for visual indent
sympy/polys/ring_series.py:1516:21: E128 continuation line under-indented for visual indent
sympy/polys/ring_series.py:1527:1: E302 expected 2 blank lines, found 1
sympy/polys/ring_series.py:1554:1: E302 expected 2 blank lines, found 1
sympy/polys/ring_series.py:1581:1: E302 expected 2 blank lines, found 1
sympy/polys/ring_series.py:1602:1: E302 expected 2 blank lines, found 1
sympy/polys/ring_series.py:1639:21: E128 continuation line under-indented for visual indent
sympy/polys/ring_series.py:1645:21: E128 continuation line under-indented for visual indent
sympy/polys/ring_series.py:1656:1: E302 expected 2 blank lines, found 1
sympy/polys/ring_series.py:1678:1: E302 expected 2 blank lines, found 1
sympy/polys/ring_series.py:1708:1: E302 expected 2 blank lines, found 1

**Comment by leosartaj** [comment]
Anyways, it's up to you. 

+1

**Comment by shivamvats** [comment]
Thanks! I'll try to improve it. :)

**Comment by shivamvats** [comment]
@leosartaj I hope this is better.

**Comment by leosartaj** [comment]
+1, if the tests pass.

**Comment by certik** [comment]
Tests pass, merging.

**Comment by certik** [comment]
Tests pass, merging.

**Comment by shivamvats** [comment]
Thanks!

File: sympy/polys/ring_series.py [hunk 3]
```diff
@@ -1,19 +1,15 @@
-from sympy.polys.domains import QQ, EX, ExpressionDomain
+from sympy.polys.domains import QQ, EX
 from sympy.polys.rings import PolyElement, ring
 from sympy.polys.polyerrors import DomainError
 from sympy.polys.monomials import (monomial_min, monomial_mul, monomial_div,
-    monomial_ldiv)
+                                   monomial_ldiv)
```
Comment (toolforger) [inline sympy/polys/ring_series.py]: 4 spaces of indent are generally okay for us, so this is not necessary.
In face we have a preference of 4 spaces over hanging indent, so this would be marginally worse.

File: sympy/polys/ring_series.py [hunk 4]
```diff
@@ -101,18 +93,19 @@ def rs_is_puiseux(p, x):
...
     """
-    ii = p.ring.gens.index(x)
-    n = 1
+    index = p.ring.gens.index(x)
     for k in p:
-        if k[ii] != int(k[ii]):
+        if k[index] != int(k[index]):
             return True
-        if k[ii] < 0:
+        if k[index] < 0:
```
Comment (toolforger) [inline sympy/polys/ring_series.py]: The above change is semantic, not merely PEP8.
It's better to keep PEP8 corrections separate from semantic changes. People who're interested in a semantic change won't have to sift through a pile of PEP8 differences to find what they're after.

Comment (toolforger) [inline sympy/polys/ring_series.py]: Setting up the PR so that it is two commits (one for PEP8, one for semantics) is enough, because people are visiting the commit log and are not overly concerned with PRs. Just make sure that each commit has the appropriate comment.

File: sympy/polys/ring_series.py [hunk 5]
```diff
@@ -101,18 +93,19 @@ def rs_is_puiseux(p, x):
...
             raise ValueError('The series is not regular in %s' % x)
     return False
 
 def rs_puiseux(f, p, x, prec):
     """
-    Return the puiseux series for `f(p, x, prec)` when `f` is implemented only for regular series
+    Return the puiseux series for `f(p, x, prec)`
```
Comment (toolforger) [inline sympy/polys/ring_series.py]: That's missing a full stop (.) at the end.

File: sympy/polys/ring_series.py [hunk 6]
```diff
@@ -101,18 +93,19 @@ def rs_is_puiseux(p, x):
...
+
+    To be used when function`f` is implemented only for regular series
```
Comment (toolforger) [inline sympy/polys/ring_series.py]: Full stop missing.

File: sympy/polys/ring_series.py [hunk 7]
```diff
@@ -137,26 +129,27 @@ def rs_puiseux(f, p, x, prec):
...
     else:
         r = f(p, x, prec)
     return r
 
 def rs_puiseux2(f, p, q, x, prec):
     """
-    Return the puiseux series for `f(p, q, x, prec)` when `f` is implemented only for regular series
+    Return the puiseux series for `f(p, q, x, prec)`
 
+    To be used when function `f` is implemented only for regular series
```
Comment (toolforger) [inline sympy/polys/ring_series.py]: Two full stops missing.

File: sympy/polys/ring_series.py (hunk 4 above)
Comment (toolforger) [inline sympy/polys/ring_series.py]: BTW renaming `ii` to `index` is pointless.
Also, funny as it may sound, `index` isn't a good name for an index in code, idiomatic (and instantly understood) names for index variables are `i`, `j`, `k` etc. `index` carries a hint that the code is talking about an index in the application domain. Since we do have index as a domain term (Einstein indexes and such), this is actually actively confusing, so I recommend renaming `ii` to `i` (unless `i` is already taken).

Comment (toolforger) [inline sympy/polys/ring_series.py]: Oh. Wait. We *are* talking about a domain object here, are we?

File: sympy/polys/ring_series.py [hunk 8]
```diff
@@ -368,8 +360,8 @@ def rs_subs(p, rules, x, prec):
                 if r == 0 and (i, q) in d:
                     d[(i, power)] = rs_square(d[(i, q)], x, prec)
                 elif (i, power - 1) in d:
-                    d[(i, power)] = rs_mul(d[(i, power - 1)], d[(i, 1)], \
-                        x, prec)
+                    d[(i, power)] = rs_mul(d[(i, power - 1)], d[(i, 1)],
+                                           x, prec)
```
Comment (toolforger) [inline sympy/polys/ring_series.py]: No hanging indent here.
I tend to add consistent line breaks, giving me

```
                    d[(i, power)] = rs_mul(
                        d[(i, power - 1)],
                        d[(i, 1)],
                        x,
                        prec)
```

Since the argument list isn't *that* long, this might be an option, too:

```
                    d[(i, power)] = rs_mul(
                        d[(i, power - 1)], d[(i, 1)], x, prec)
```

I prefer breaking at the outermost precedence level, which would be around the assignment operator:

```
                    d[(i, power)] = \
                        rs_mul(d[(i, power - 1)], d[(i, 1)], x, prec)
```

I'm not too confident that this last variant is the best option; the rule is to double-check for readability after formatting changes.

File: sympy/polys/ring_series.py [hunk 9]
```diff
@@ -439,10 +431,10 @@ def _series_inversion1(p, x, prec):
         prec = int(prec)
 
     if zm not in p:
-        raise ValueError('no constant term in series')
+        raise ValueError("No constant term in series")
     if _has_constant_term(p - c, x):
-        raise ValueError('p cannot contain a constant term depending on '
-            'parameters')
+        raise ValueError("p cannot contain a constant term depending on "
+                         "parameters")
```
Comment (toolforger) [inline sympy/polys/ring_series.py]: No hanging indents please.

File: sympy/polys/ring_series.py [hunk 10]
```diff
@@ -477,32 +468,30 @@ def rs_series_inversion(p, x, prec):
...
-
 def _coefficient_t(p, t):
-    """
-    Coefficient of `x_i**j` in p, where t = (i, j)
-    """
+    """Coefficient of `x_i**j` in p, where t = (i, j)"""
     i, j = t
-    ring = p.ring
-    expv1 = [0]*ring.ngens
+    R = p.ring
```
Comment (toolforger) [inline sympy/polys/ring_series.py]: Why did you rename `ring` to `R`?
Not that I'm against renaming away from `ring`, but `R` now tells the reader less about what's going on than before.

File: sympy/polys/ring_series.py [hunk 11]
```diff
@@ -555,20 +544,19 @@ def rs_series_reversion(p, x, n, y):
...
-    ny = ring.gens.index(y)
+    R = p.ring
+    nx = R.gens.index(x)
+    y = R(y)
+    ny = R.gens.index(y)
     if _has_constant_term(p, x):
-        raise ValueError('p must not contain a constant term in the series \
-            variable')
+        raise ValueError("p must not contain a constant term in the series "
+                         "variable")
```
Comment (toolforger) [inline sympy/polys/ring_series.py]: No hanging indent please.

File: sympy/polys/ring_series.py [hunk 12]
```diff
@@ -729,8 +717,9 @@ def rs_fun(p, f, *args):
     of a multivariate series:
 
       rs_fun(p, tan, iv, prec)
-      tan series is first computed for a dummy variable _x, ie, tan(_x, iv, prec)
-      Then we substitute _x with p to get the desired series
+      tan series is first computed for a dummy variable _x,
+      ie, rs_tan(_x, iv, prec). Then we substitute _x with p to get the
```
Comment (toolforger) [inline sympy/polys/ring_series.py]: `ie` -> `i.e.` (I think)

File: sympy/polys/ring_series.py [hunk 13]
```diff
@@ -897,14 +884,14 @@ def rs_nth_root(p, n, x, prec):
                 c_expr = c.as_expr()
                 const = R(c_expr**(QQ(1, n)))
             except ValueError:
-                raise DomainError("The given series can't be expanded in this "
-                    "domain.")
+                    raise DomainError("The given series can't be expanded in "
```
Comment (toolforger) [inline sympy/polys/ring_series.py]: Four spaces for indent, not eight.

File: sympy/polys/ring_series.py [hunk 14]
```diff
@@ -897,14 +884,14 @@ def rs_nth_root(p, n, x, prec):
...
+                                      "this domain.")
```
Comment (toolforger) [inline sympy/polys/ring_series.py]: No hanging indent please.

File: sympy/polys/ring_series.py [hunk 15]
```diff
@@ -897,14 +884,14 @@ def rs_nth_root(p, n, x, prec):
...
-            try:                             # RealElement doesn't support
-                const = R(c**Rational(1, n)) # exponentiation with mpq object
-            except ValueError:               # as exponent
-                raise DomainError("The given series can't be expanded in this "
-                "domain")
+            try:                              # RealElement doesn't support
+                const = R(c**Rational(1, n))  # exponentiation with mpq object
+            except ValueError:                # as exponent
+                    raise DomainError("The given series can't be expanded in "
+                                      "this domain.")
```
Comment (toolforger) [inline sympy/polys/ring_series.py]: See above.

File: sympy/polys/ring_series.py [hunk 16]
```diff
@@ -955,14 +942,14 @@ def rs_log(p, x, prec):
                 try:
                     const = R(log(c_expr))
                 except ValueError:
-                    raise DomainError("The given series can't be expanded in this "
-                        "domain.")
+                    raise DomainError("The given series can't be expanded in "
+                                      "this domain.")
```
Comment (toolforger) [inline sympy/polys/ring_series.py]: Okay, not going to comment on the remaining hanging indents.

File: sympy/polys/ring_series.py (hunk 13 above)
Comment (shivamvats) [inline sympy/polys/ring_series.py]: This is a mistake.

File: sympy/polys/ring_series.py (hunk 10 above)
Comment (shivamvats) [inline sympy/polys/ring_series.py]: There is a `ring` function too. We might want to use that inside the function.

File: sympy/polys/ring_series.py (hunk 4 above)
Comment (shivamvats) [inline sympy/polys/ring_series.py]: It is the index of the generator in the ring.

File: sympy/polys/ring_series.py (hunk 9 above)
Comment (shivamvats) [inline sympy/polys/ring_series.py]: I think it is very subjective. Personally, I find it more readable this way.

Comment (toolforger) [inline sympy/polys/ring_series.py]: Readability does have subjective elements, but it's not just subjectivity.
General consensus has been leaning towards multiples-of-four indents, not hanging indent. Not just in SymPy, but in the Python community in general, though of course everybody's perception tends to be different due to different peer groups. All I can say that in the discussions I've seen on the topic, more often than not, hanging indents are considered an undesirable option, some people even consider their inclusion in PEP8 a mistake; more importantly, I see little if any protest against that stance. In fact you're the first person I ever met who expressed a strong preference for hanging indents.

A more tangible issue with hanging indents is that if the name changes, you have to touch not just the lines with the name but also the names with the parameters on followup lines.

Also, there's the point to be made that subjective preferences shouldn't lead to changes in the code where it already follows another style.

Comment (shivamvats) [inline sympy/polys/ring_series.py]: @toolforger 
> Not just in SymPy, but in the Python community in general, though of course everybody's >perception tends to be different due to different peer groups. All I can say that in the >discussions I've seen on the topic, more often than not, hanging indents are considered an >undesirable option, some people even consider their inclusion in PEP8 a mistake; more >importantly, I see little if any protest against that stance. In fact you're the first person I ever >met who expressed a strong preference for hanging indents.

I'm afraid I had assumed the opposite as I was asked to change from multiple-of-four indent to hanging indent in this PR (see the outdated diff above). That said, I don't have a strong preference for either of them. There is too much variety in coding styles (including hanging indents) even in Sympy.

Since this PR is already merged, you can make the changes in another PR if you want.

Comment (toolforger) [inline sympy/polys/ring_series.py]: > I'm afraid I had assumed the opposite as I was asked to change from multiple-of-four indent to hanging indent in this PR (see the outdated diff above).

I see. I guess @leosartaj is on a different boat with that one.

 > There is too much variety in coding styles (including hanging 
indents) even in Sympy.

That's true.

> Since this PR is already merged,

Yeah, I noticed too late.

 > you can make the changes in another PR if you want.

That would come as a big style clean-up, not in a piecemeal fashion :-)
The important point is that people know so the style is somewhat more 
consistent over time.

Comment (shivamvats) [inline sympy/polys/ring_series.py]: Cool!

Comment (leosartaj) [inline sympy/polys/ring_series.py]: > I see. I guess @leosartaj is on a different boat with that one.
There is too much variety in coding styles (including hanging
indents) even in Sympy.

I am not an expert on PEP8 or other coding styles. I started using flake8 to solve the issue. It complained a lot about lines being under-indented or over-indented in some cases. So the changes. Well, it can be decided to not use hanging indents and go for 4-indents.

Comment (certik) [inline sympy/polys/ring_series.py]: These minor indentation issues are really not a big deal. Each person has a different preference, and one just needs to get used to that other people might format things a bit differently, and not lose too much time discussing it. So I agree with other things that @toolforger proposed (like missing `.` or in other PRs missing space around operators like `==`), but do not lose time worrying about how to properly indent multiline command.

Comment (toolforger) [inline sympy/polys/ring_series.py]: > I started using flake8 to solve the issue. It complained a lot

Yeah, automatic checkers complain a lot.
I guess the best approach to automatic code checkers is to start with 
all checks switched off and explicitly enabling only those that make 
sense for a project.

**Comment by pernici** [comment]
There is a bug in `rs_sin`; 

```
-    if len(p) > 20 and p.ngens == 1:
+    if len(p) > 20 and R.ngens == 1:
```

**Comment by shivamvats** [comment]
@pernici I think this is fixed.
